docs/usage/routing/index.rst

```diff
@@ -1,9 +1,412 @@
 Routing
 =======
 
+Litestar's routing system is based around a few basic components:
 
-.. toctree::
+Route handlers
+    Callables that receive parameters from the request or dependencies and return
+    response data
+
+Controllers
+    Classes that group together multiple route handlers and define common attributes
+    such as dependencies and middlewares
+
+Routers
+    A lightweight application on which route handlers, controllers and other routers can
+    be registered, as well as common configurations such as dependencies and middlewares
+
+Applications
+    Instances of :class:`~litestar.app.Litestar`, a special type of router that serves
+    as the entry point of an application, in addition to receive the same configuration
+    routers can
```

Comment:
```suggestion
    Instances of :class:`~litestar.app.Litestar`, the entry point of an application, and a 
    specialized type of router, so it can receive the same configuration as routers can
```
This wording was a bit awkward - not sure if my suggestion is much better though!